src/test/java/uk/gov/hmcts/reform/blobrouter/servicebus/notifications/NotificationsPublisherTest.java

```diff
@@ -0,0 +1,102 @@
+package uk.gov.hmcts.reform.blobrouter.servicebus.notifications;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.QueueClient;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import uk.gov.hmcts.reform.blobrouter.servicebus.notifications.model.NotificationMsg;
+import uk.gov.hmcts.reform.blobrouter.util.MessageBodyRetriever;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class NotificationsPublisherTest {
+
+    private NotificationsPublisher notifier;
+
+    @Mock
+    private QueueClient queueClient;
+
+    @BeforeEach
+    void setUp() {
+        notifier = new NotificationsPublisher(queueClient, new ObjectMapper());
+    }
+
+    @Test
+    void notify_should_send_message_with_right_content() throws Exception {
+        // given
+        NotificationMsg notificationMsg = new NotificationMsg(
+            "test.zip",
+            "ABC",
+            "PO123",
+            "1234",
+            "Invalid Signature",
+            "Signature verification failed",
+            "blob-router"
+        );
+
+        // when
+        notifier.publish(notificationMsg);
+
+        // then
+        ArgumentCaptor<Message> messageCaptor = ArgumentCaptor.forClass(Message.class);
+        verify(queueClient).send(messageCaptor.capture());
+
+        Message message = messageCaptor.getValue();
+
+        assertThat(message.getContentType()).isEqualTo("application/json");
+
+        String messageBodyJson = new String(MessageBodyRetriever.getBinaryData(message.getMessageBody()));
```

Comment:
If MessageBodyRetriever is not supposed to be used in different test classes it makes sense to move getBinaryData into this class as a method

Reply:
Moved to the test class